lib/AST/IRToASTVisitor.cpp

```diff
@@ -398,8 +398,28 @@ void IRToASTVisitor::VisitArgument(llvm::Argument &arg) {
   // Get parent function declaration
   auto func{arg.getParent()};
   auto fdecl{clang::cast<clang::FunctionDecl>(GetOrCreateDecl(func))};
+  auto argtype{arg.getType()};
+  if (arg.hasByValAttr()) {
+    argtype = llvm::cast<llvm::PointerType>(argtype)->getElementType();
+  }
   // Create a declaration
-  parm = ast.CreateParamDecl(fdecl, GetQualType(arg.getType()), name);
+  parm = ast.CreateParamDecl(fdecl, GetQualType(argtype), name);
+}
+
+static llvm::FunctionType *GetFixedFunctionType(llvm::Function &func) {
```

Comment:
Add a comment with a bit of an example of what `byval` arguments look like, and how we re-interpret them as actual values.